scrapy/mail.py

```diff
@@ -89,7 +89,7 @@ def send(self, to, subject, body, cc=None, attachs=(), mimetype='text/plain', ch
                 part.add_header('Content-Disposition', 'attachment', filename=attach_name)
                 msg.attach(part)
         else:
-            msg.set_payload(body)
+            msg.set_payload(body, charset)
```

Comment:
Looking at https://bugs.python.org/issue31820, I wonder if the solution is instead to move the 2 charset lines to _after_ the body or attachments are set, and make no change here.

As for tests, it seems `get_payload` has `decode=False` by default, so it should return the encoded body.

Reply:
thanks @Gallaecio for your suggestions. I tried to add set_charset after set_payload, updating tests to reflect changes but again encoded body and attaches is not visible in tests.